sabnzbd/bpsmeter.py

```diff
@@ -204,9 +205,21 @@ def read(self):
             self.update()
         return res
 
-    def update(self, server: Optional[str] = None, amount: int = 0):
+    def update(self, server: Optional[str] = None, amount: int = 0, force: bool = False):
         """ Update counters for "server" with "amount" bytes """
         t = time.time()
+
+        # Add amount to temporary storage
+        if server:
+            if server not in self.temp_amount:
+                self.temp_amount[server] = 0
+                self.server_bps[server] = 0.0
+            self.temp_amount[server] += amount
+
+        # Wait at least 0.021 seconds between each full update
+        if not force and t - self.last_update < 0.021:
```

Comment:
I think I need a bit more details why this _very_ specific number of seconds? 😋

Reply:
Yes, that may not have been necessary. I wanted it to be more than 0.02 and the Windows internal clock is a bit kinky and has a tendency to be a few nanoseconds off a round number.

Reply:
What happens if you use something like 0.5 or 0.1?
Probably also reduces CPU usage.

Reply:
I tested setting it to `1` and it still works fine, visually. At most it's 1 second delayed, seems fine by me,
Maybe that increases the reduction of time spent in this function?

Reply:
I'm worried that a too long delay could make the transfer rate erratic for those who use the bps limiter. If it's not updated for 1 second then the transfer rate could speed up too much. Then it would sleep for too long to even it out, which would slow it down too much. 0.021 seconds seems safe, and reduces the number of full updates from potentially tens of thousands per second to less than 100.